src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -0,0 +1,250 @@
+    }
+
+    @Override
+    public Statement createStatement(int resultSetType, int resultSetConcurrency) throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-16");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int resultSetType, int resultSetConcurrency)
+            throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-13");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, int autoGeneratedKeys) throws SQLException {
+        throw new UnsupportedOperationSQLException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String mql, String[] columnNames) throws SQLException {
+        throw new UnsupportedOperationSQLException("Auto-generated key from MongoDB server side unsupported");
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // SQL99 data types
+
+    @Override
+    public Clob createClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Blob createBlob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public NClob createNClob() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public SQLXML createSQLXML() throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    @Override
+    public Struct createStruct(String typeName, Object[] attributes) throws SQLException {
+        throw new NotYetImplementedSQLException();
+    }
+
+    // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Database meta data
+
+    @Override
+    public DatabaseMetaData getMetaData() throws SQLException {
+        throw new NotYetImplementedSQLException(
+                "To be implemented in scope of https://jira.mongodb.org/browse/HIBERNATE-37");
+    }
+
+    /**
+     * Used during Hibernate's DDL step for Information Extraction purposes.
+     *
+     * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
+     */
+    @Override
+    public @Nullable String getCatalog() {
+        return null;
+    }
+
+    /**
+     * Used during Hibernate's DDL step for Information Extraction purposes.
+     *
+     * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
+     */
+    @Override
+    public @Nullable String getSchema() {
+        return null;
+    }
+
```

Comment:
I spent most of research time for the above two methods implication. From my current digging, they are only used for Hibernate's DDL step. See https://docs.jboss.org/hibernate/orm/6.6/userguide/html_single/Hibernate_User_Guide.html#schema-generation for the relevant content. Hibernate supports the `hibernate.hbm2ddl.auto` property with `create` option value, so Hibernate has nontrivial code logic to auto-generate DDL scripts and the above two methods are meant for this purposes.

As a NoSQL (no negative connotation) DB, it seems irrelevant for we have no DDL step except for indexing and unique constraint creation task (in Milestone 4 scope).